### DISCUSSION TITLE
No action detected in task --- WAF install failing

### DISCUSSION THREAD

**Discussion by old-hpux** [roxy-wi/roxy-wi#322] [General]
Attempted to install WAF via the gui or command line and fails with same message:

user@hostname:/var/www/haproxy-wi/app$ sudo /bin/bash scripts/waf.sh PROXY= HAPROXY_PATH=/etc/haproxy VERSION=1.8.8-1ubuntu0.11 SSH_PORT=22 HOST=X.X.X.X USER=uuser PASS=None KEY=/var/www/haproxy-wi/keys/Myserver_Default.pem
ERROR! no action detected in task. This often indicates a misspelled module name, or incorrect module path.

The error appears to have been in '/var/www/haproxy-wi/app/scripts/ansible/roles/waf/tasks/main.yml': line 370, column 7, but may
be elsewhere in the file depending on the exact syntax problem.

The offending line appears to be:

  always:
    - name: Remove modsecurity.tar.gz
      ^ here

error: Cannot install WAF

**Comment by Aidaho12** [comment]
it looks like you server does not have the Internet connection and cannot download ModSec. Try to download ModSec

**Reply by old-hpux** [reply]
the server has direct internet connection and I was able to download modsec this way:
wget https://github.com/SpiderLabs/ModSecurity/releases/download/v2.9.5/modsecurity-2.9.5.tar.gz

but the question comes, how do I install this?  untar to a particular location?

**Reply by Aidaho12** [reply]
Unfortunately, it will not help. What ansible version do you use?

**Reply by old-hpux** [reply]
$ ansible --version
ansible 2.5.1
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/home/ubuntu/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/dist-packages/ansible
  executable location = /usr/bin/ansible
  python version = 2.7.17 (default, Mar 18 2022, 13:21:42) [GCC 7.5.0]

**Reply by Aidaho12** [reply]
Pretty old version, update it to 2.12 at least

**Reply by old-hpux** [reply]
Thank you, That fixed it.  As I'm on Bionic 18.04, the latest is 2.9.27, but it worked!

I must have missed the version requirement in the install docs.  Thanks for your guidance!
